central/componentcveedge/datastore/internal/postgres/store_test.go

```diff
@@ -56,7 +56,7 @@ func (s *ImageComponentCveRelationsStoreSuite) TestStore() {
 	componentCVEEdge := &storage.ComponentCVEEdge{}
 	s.NoError(testutils.FullInit(componentCVEEdge, testutils.SimpleInitializer(), testutils.JSONFieldsFilter))
 
-	foundComponentCVEEdge, exists, err := store.Get(ctx, componentCVEEdge.GetId(), componentCVEEdge.GetImageComponentId(), componentCVEEdge.GetCveId())
+	foundComponentCVEEdge, exists, err := store.Get(ctx, componentCVEEdge.GetId(), componentCVEEdge.GetImageComponentId(), componentCVEEdge.GetImageComponentName(), componentCVEEdge.GetImageComponentVersion(), componentCVEEdge.GetImageComponentOperatingSystem(), componentCVEEdge.GetImageCveId(), componentCVEEdge.GetImageCve(), componentCVEEdge.GetImageCveOperatingSystem())
```

Comment:
Hmm is this correct that it has operating system twice?

Reply:
This is the one pondered on for a while. We know that operating system is available with components as well as cves. By defining an explicit fk in relations, for component and cve operating system field, the auto-generation of joins and schema remained straightforward. Maybe there is a way to avoid it, but it is going to be conditional.